packages/@aws-cdk/aws-apigateway/lib/authorizers/lambda.ts

```diff
@@ -90,6 +93,34 @@ abstract class LambdaAuthorizer extends Authorizer implements IAuthorizer {
     }
 
     this.restApiId = restApi.restApiId;
+
+    let functionName;
+
+    if (this.handler instanceof lambda.Function) {
+      // if not imported, extract the name from the CFN layer to reach
+      // the literal value if it is given (rather than a token)
+      functionName = (this.handler.node.defaultChild as lambda.CfnFunction).functionName;
+    } else {
+      // imported, just take the function name.
+      functionName = this.handler.functionName;
+    }
+
+    let authorizerToken;
+
+    if (!Token.isUnresolved(functionName)) {
+      authorizerToken = JSON.stringify({ functionName });
+    }
+
+    const addToLogicalId = FeatureFlags.of(this).isEnabled(APIGATEWAY_AUTHORIZER_CHANGE_DEPLOYMENT_LOGICAL_ID);
+
+    const deployment = restApi.latestDeployment;
+    if (deployment && addToLogicalId) {
+      deployment.node.addDependency(this);
+      deployment.addToLogicalId({
+        authorizer: this.authorizerProps,
```

Comment:
Is the `authorizerToken` not enough?

Reply:
I think we need all of them; if any of the authorizer properties change, we need a new deployment, otherwise the changes will not be reflected in the API